### PR TITLE
Sundeep Agarwal's one liner article

### PR DESCRIPTION
**Pull request by briandfoy**
What we need while we process this while we edit:

- [x] author bio in _data/authors/sundeep-agarwal_
- [x] author pic in _static/images/author/sundeep-agarwal.jpg_

### CODE REVIEW

**Comment by dnmfarrell** [comment]
This has changes to these files which need to be removed, maybe rebase the branch onto master?

    content/article/closures-as-objects.md
    layouts/shortcodes/mcpan.html
    static/images/author/david-farrell.jpg

File: content/article/closures-as-objects.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+```perl
+my $class_point = Class::Lambda::new_class(
+  'Point',
+  {
+    x => undef,
+    y => undef,
+  },
+  {
+    x => sub { $_[0]->{x} },
+    y => sub { $_[0]->{y} >}});
```
[the remark this answers is not in the document]
Comment (briandfoy) [reply content/article/closures-as-objects.md:127]: Hey Josh,

Thanks for looking at these things! But, just to save your time, note that this is a work in progress and we're still editing it. We'll definitely look at all the things you note.

File: content/article/perl-one-liners.md [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+
+
+#######################
+
+```bash
+perl -0777 -ne '$#ARGV==1 ? $s=$_ : $#ARGV==0 ? $r=$_ :
+                print s/\Q$s/$r/gr' search.txt replace.txt ip.txt
+```
+
+Unlike grep, sed, or Awk, Perl doesn't remove the input record by default, so the last item has the newline. I have this
```
[the remark this answers is not in the document]
Comment (briandfoy) [reply content/article/perl-one-liners.md:137]: > This has changes to these files which need to be removed, maybe rebase the branch onto master?

huh, that's probably my fault. When I next work on this branch I'll fix that. I'll probably cherry-pick commits into two new branches, commit the small changes, and rebase this off master.

[the remark this answers is not in the document]
Comment (briandfoy) [reply content/article/perl-one-liners.md:137]: Huh, this was a reply to David later in the conversation

**Comment by briandfoy** [comment]
Somehow my reply to David was applied to a different conversation. Next time I work on this branch I'll take care of the extra commits.

File: content/article/perl-one-liners.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (learnbyexample) [reply content/article/perl-one-liners.md:137]: >This line feels like it lost its end

I did catch that during the second round and there's lot of changes I made that isn't reflected here yet.

[the remark this answers is not in the document]
Comment (briandfoy) [reply content/article/perl-one-liners.md:137]: Just checking on the status of your changes. Is there something that you have to change in the article?

[the remark this answers is not in the document]
Comment (learnbyexample) [reply content/article/perl-one-liners.md:137]: @briandfoy I went through the revised article again, I don't have further changes. I'd need your input if changes are to be made based on some points I had mentioned in the last mail.

**Pull request closed without merging**